### PR TITLE
fix: Don't clober event group when renaming vars

### PR DESCRIPTION
**Pull request by NeilFraser**
Also audit all existing event group commands and tweak a few of them where I think there's a potential issue.

See https://groups.google.com/g/blockly/c/dKizQbKb8vo

### CODE REVIEW

File: core/contextmenu_items.ts [hunk 1]
```diff
@@ -279,7 +278,8 @@ export function registerDeleteAll() {
       }
       scope.workspace.cancelCurrentGesture();
       const deletableBlocks = getDeletableBlocks_(scope.workspace);
-      const eventGroup = idGenerator.genUid();
+      eventUtils.setGroup(true);
```
Comment (gonfunko) [inline core/contextmenu_items.ts:281]: If the user chooses cancel on the dialog, deleteNext_ is never run and grouping is never turned off. deleteNext_ was already calling setGroup() with the passed-in ID - why is this change necessary?

File: core/workspace_svg.ts [hunk 2]
```diff
@@ -1253,21 +1253,22 @@ export class WorkspaceSvg extends Workspace implements IASTNodeLocationSvg {
...
-
     let pastedThing;
-    // Checks if this is JSON. JSON has a type property, while elements don't.
-    if (state['type']) {
-      pastedThing = this.pasteBlock_(null, state as blocks.State);
-    } else {
-      const xmlBlock = state as Element;
-      if (xmlBlock.tagName.toLowerCase() === 'comment') {
-        pastedThing = this.pasteWorkspaceComment_(xmlBlock);
+    try {
```
Comment (gonfunko) [inline core/workspace_svg.ts:1257]: What was throwing an exception here?

File: core/contextmenu_items.ts (hunk 1 above)
Comment (NeilFraser) [reply core/contextmenu_items.ts:281]: This changes eliminates the dependency on idGenerator.  It also means that in Blockly there would be only one place which sets eventGroup -- so if we change the group format in the future, we won't need to change it in two places.

File: core/workspace_svg.ts (hunk 2 above)
Comment (NeilFraser) [reply core/workspace_svg.ts:1257]: If the clipboard is corrupted, the XML parser might throw.

**Comment by gonfunko** [comment]
Does this also address #1534?

**Review by gonfunko — APPROVED** [review]
